### PR TITLE
fix: change UI when restricting to only one country on PhoneInput

### PR DESCRIPTION
**Pull request by gtallesb**
fix: #1649 

## Changes proposed in this PR:
-add new conditional component VariantRenderIf
-replace one of the RenderIf components with VariantRenderIf on index.js
-add a new interactive example passing the countries prop on .md

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/PhoneInput/index.js [hunk 1]
```diff
@@ -140,22 +139,18 @@ const PhoneInput = React.forwardRef((props, ref) => {
...
+                    ref={triggerRef}
+                    onClick={handleClick}
+                    onFocus={event => handleFocus(event, 0)}
+                    onBlur={handleBlur}
+                    tabIndex={tabIndex}
+                    disabled={disabled}
+                    error={error}
+                    countries={countriesProps}
+                    flagIcon={flagIcon}
+                />
```
Comment (yvmunayev) [inline src/components/PhoneInput/index.js:153]: I think the solution can be simpler something like this:

```js
    const isReadOnly = readOnly && !disabled;
    const hasTrigger = !isReadOnly && countries.length > 1;
    const isOnlyCountry = !isReadOnly && countries.length === 1;
    
    
    
    <RenderIf isTrue={isReadOnly}>
        <StyledFlagContainer readOnly>{flagIcon}</StyledFlagContainer>
    </RenderIf>

    <RenderIf isTrue={hasTrigger}>
        <StyledTrigger
            ref={triggerRef}
            onClick={handleClick}
            onFocus={event => handleFocus(event, 0)}
            onBlur={handleBlur}
            tabIndex={tabIndex}
            disabled={disabled}
        >
            <StyledFlagContainer disabled={disabled}>
                {flagIcon}
                <StyledIndicator error={error} disabled={disabled} />
            </StyledFlagContainer>
        </StyledTrigger>
    </RenderIf>

    <RenderIf isTrue={isOnlyCountry}>
        <StyledFlagContainer disabled={disabled}>{flagIcon}</StyledFlagContainer>
    </RenderIf>
```

File: src/components/PhoneInput/readme.md [hunk 2]
```diff
@@ -29,3 +29,36 @@ const Form = () => {
...
+const Form = () => {
+    const [phone, setPhone] = useState();
+
+    return (
+        <PhoneInput
+            label="Phone Number"
+            placeholder="Enter your phone number"
+            onChange={setPhone}
+            value={phone}
+            countries=''
```
Comment (yvmunayev) [inline src/components/PhoneInput/readme.md:55]: countries has to be an array

File: src/components/PhoneInput/readme.md [hunk 3]
```diff
@@ -29,3 +29,38 @@ const Form = () => {
...
+            
+        />
+    );
+};
+
+<Container>
+    <Form />
+</Container>
+
+```
```
Comment (reiniergs) [inline src/components/PhoneInput/readme.md:66]: remove unnecessary empty lines ^^

File: src/components/PhoneInput/readme.md [hunk 4]
```diff
@@ -29,3 +29,38 @@ const Form = () => {
...
+const Form = () => {
+    const [phone, setPhone] = useState();
+
+    return (
+        <PhoneInput
+            label="Phone Number"
+            placeholder="Enter your phone number"
+            onChange={setPhone}
+            value={phone}
+            countries={[]}
```
Comment (reiniergs) [inline src/components/PhoneInput/readme.md:55]: it's an example with a single selection, right?? I see empty array []

Comment (TahimiLeonBravo) [reply src/components/PhoneInput/readme.md:55]: I think when the `countries` receive an empty array it should render all the countries

**Review by TahimiLeonBravo — CHANGES_REQUESTED** [review]

File: src/components/PhoneInput/readme.md [hunk 5]
```diff
@@ -29,3 +29,38 @@ const Form = () => {
...
+
+    return (
+        <PhoneInput
+            label="Phone Number"
+            placeholder="Enter your phone number"
+            onChange={setPhone}
+            value={phone}
+            countries={['ca']}
+            
+
```
Comment (maxxgreene) [inline src/components/PhoneInput/readme.md:57]:
```suggestion
```

**Review by maxxgreene — APPROVED** [review]
+1

**Pull request merged**